### DISCUSSION TITLE
[firebase_auth] Add support for linkWithPopup and linkWithRedirect

### DISCUSSION THREAD

**Discussion by doppio** [firebase/flutterfire#6070] [Feature request] [answered]
The auth package supports linking Firebase accounts via credential and phone number, but not via the pop-up based flow, which is preferred on web. The Firebase User API has a [linkWithPopup](https://firebase.google.com/docs/reference/js/firebase.User#linkwithpopup) function which I presume could be exposed through FlutterFire.

My use case is that I have enabled the "One account per email address" setting in my Firebase project. The FlutterFire documentation [explains](https://firebase.flutter.dev/docs/auth/error-handling#handling-account-exists-with-different-credential-errors) how to handle "account-exists-with-different-credential" errors when using  the recommended native flow, but an equivalent approach doesn't seem to be possible exclusively using the popup flow on web.

The same can be said for `linkWithRedirect`.

**Comment by rbozan** [comment]
Same goes for `linkWithRedirect`

**Reply by doppio** [reply]
Thanks, that's a good point. I've updated the discussion to mention that.

**Comment by doppio** [comment, marked as answer]
As @darshankawar pointed out to me in #7360, `linkWithPopup` is now supported (see PR #6562).
